python/paddle/optimizer/optimizer.py

```diff
@@ -1021,16 +1053,25 @@ def clear_grad(self):
                 adam.clear_grad()
 
         """
+        param_list = []
         if self._parameter_list is None or not isinstance(
                 self._parameter_list[0], dict):
             for p in self._parameter_list:
                 if not p.stop_gradient:
-                    p.clear_gradient()
+                    if set_to_zero:
+                        p.clear_gradient()
+                    else:
+                        param_list.append(p)
         else:
             for param_group in self._param_groups:
                 for p in param_group['params']:
                     if not p.stop_gradient:
-                        p.clear_gradient()
+                        if set_to_zero:
+                            p.clear_gradient()
+                        else:
+                            param_list.append(p)
```

Comment:
I think we can use `core.clear_gradients` even if `set_to_zero is true`

Reply:
Done, tks.